build/helper/metadata_find.py

```diff
@@ -5,11 +5,17 @@ def find_parameter(name, parameters):
     return parameter[0]
 
 
-def find_size_parameter(parameter, parameters):
+def find_size_parameter(parameter_list, parameters):
     '''Returns the parameter that is used to specify the size of another parameter. Applies to 'ivi-dance' and 'passed-in'.'''
-    if not parameter:
+    if len(parameter_list) == 0:
         return None
-    return find_parameter(parameter['size']['value'], parameters)
+    # We verified earlier that all paramers have the same size parameter, so we only need to use the first one
```

Comment:
"earlier" implies that you know the order in which this is called. You should say simply that it's an assumption.

Reply:
Changed to:
`Assumption: all paramers have the same size parameter, so we only need to use the first one`